### PR TITLE
feat: LT-1268 - case advanced search

### PR DESCRIPTION
**Pull request by a-bek**

### CODE REVIEW

File: queues/views.py [hunk 1]
```diff
@@ -79,7 +76,37 @@ def get(self, request, **kwargs):
...
+                TextInput(name="party_name", title="party name"),
+                TextInput(name="party_address", title="party address"),
+                TextInput(name="goods_related_description", title="goods related description"),
+                AutocompleteInput(name="country", title="country", options=get_countries(request, convert_to_options=True)),
+                AutocompleteInput(
+                    name="control_list_entry",
+                    title="control list entry",
+                    options=get_control_list_entries(request, convert_to_options=True),
+                ),
+                TokenBar(name="flags", title="flags", input_type="token-bar", options=[
```
Comment (jfaracik) [inline queues/views.py:100]: Any chance we could get this moved out of views.py? It's rather lengthy, perhaps a filters.py?

File: cases/helpers/filters.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+def case_filters_bar(request, case_data) -> FiltersBar:
+    """
+    Returns a FiltersBar for the case search page.
+    """
+    filters = case_data["results"]["filters"]
+    statuses = [Option(option["key"], option["value"]) for option in filters["statuses"]]
+    case_types = [Option(option["key"], option["value"]) for option in filters["case_types"]]
+    gov_users = [Option(option["id"], option["full_name"]) for option in filters["gov_users"]]
+    advice_types = [Option(option["key"], option["value"]) for option in filters["advice_types"]]
+    sla_days = [Option(i, i) for i in range(99)]
```
Comment (BenAndrew-Kainos) [inline cases/helpers/filters.py:19]: Please make 99 a constant

File: cases/helpers/filters.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                case_data["results"]["is_work_queue"],
+                Checkboxes(
+                    name="hidden",
+                    options=[Option("true", CasesListPage.Filters.HIDDEN)],
+                    classes=["govuk-checkboxes--small"],
+                ),
+            ),
+        ],
+        advanced_filters=[
+            TextInput(name="exporter_application_reference", title="exporter application reference"),
```
Comment (BenAndrew-Kainos) [inline cases/helpers/filters.py:46]: all titles are content

File: cases/helpers/filters.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                title="control list entry",
+                options=get_control_list_entries(request, convert_to_options=True),
+            ),
+            TokenBar(name="flags", title="flags", options=[
+                Option(flag["id"], flag["name"])
+                for flag in get_flags(request, disable_pagination=True)
+            ]),
+            Checkboxes(
+                name="sort_by_sla_remaining",
+                options=[Option("true", "Sort by SLA remaining")],
```
Comment (BenAndrew-Kainos) [inline cases/helpers/filters.py:73]: content

File: ui_automation_tests/features/filter_cases.feature [hunk 5]
```diff
@@ -37,3 +37,18 @@ Feature: I want to filter cases on a queue
...
+    And a queue has been created
+    And I create open application or open application has been previously created
+    And case has been moved to new Queue
+    And I go to internal homepage
+    When case has been moved to new Queue
+    And I click on new queue in dropdown
+    And I show filters
+    And I click advanced filters
+    Then I can see all advanced filters
+    Then I filter by case reference
```
Comment (BenAndrew-Kainos) [inline ui_automation_tests/features/filter_cases.feature:53]: Is this sufficient e2e testing @phlptt ?

File: ui_automation_tests/features/filter_cases.feature [hunk 6]
```diff
@@ -37,3 +37,18 @@ Feature: I want to filter cases on a queue
     When I show filters
     And I click filter to show cases with open team ecju queries
     Then "1" cases are shown
+
+  @LT_1269
```
Comment (phlptt) [inline ui_automation_tests/features/filter_cases.feature:41]: need to add a tag here @ regression or @ smoke

File: ui_automation_tests/features/filter_cases.feature [hunk 7]
```diff
@@ -37,3 +37,18 @@ Feature: I want to filter cases on a queue
...
+    And I create open application or open application has been previously created
+    And case has been moved to new Queue
+    And I go to internal homepage
+    When case has been moved to new Queue
+    And I click on new queue in dropdown
+    And I show filters
+    And I click advanced filters
+    Then I can see all advanced filters
+    Then I filter by case reference
+    Then "1" cases are shown
```
Comment (phlptt) [inline ui_automation_tests/features/filter_cases.feature:54]: Then should be And

File: ui_automation_tests/features/filter_cases.feature (hunk 5 above)
Comment (phlptt) [reply ui_automation_tests/features/filter_cases.feature:53]: i think so, as discussed on call

File: ui_automation_tests/pages/case_list_page.py [hunk 8]
```diff
@@ -34,6 +35,43 @@ class CaseListPage(BasePage):
     # Queue dropdown
     LINK_CHANGE_QUEUE_ID = "link-queue"
 
+    # Advanced filters
+    CASE_REFERENCE = "case_reference"
```
Comment (phlptt) [inline ui_automation_tests/pages/case_list_page.py:39]: can you add the type of selector these are either as a comment or within the name of the selector
for example
CASE_REFERENCE_ID
or
    CASE_REFERENCE = "case_reference". # id

**Pull request merged**